src/components/PermissionsManagement/RolePermissions.jsx

```diff
@@ -41,44 +74,138 @@ function RolePermissions(props) {
     const permissionsBackEnd = permissionsObjectName.map(permission =>
       permissionFrontToBack(permission),
     );
+
+    const id = props.roleId;
+
     const updatedRole = {
-      roleName: props.role,
+      roleName: roleName,
```

Comment:
Hey Kaixiang your changes look good! But I noticed that when I updated the name of the role, it gives me a message that this role does not exist anymore, and that makes sense since we changed the name. Perhaps you could find a way to handle this and redirect to the new roleName URL.

Reply:
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/67836976/212449984-9ad500ac-c89f-410f-8d4b-262f517a5faf.png">